editoast/src/views/stdcm/stdcm_search_environment.rs

```diff
@@ -0,0 +1,243 @@
+use utoipa::ToSchema;
+
+#[cfg(test)]
+use serde::Serialize;
+
+use crate::error::Result;
+use crate::modelsv2::stdcm_search_environment::StdcmSearchEnvironment;
+use crate::modelsv2::Changeset;
+use crate::AppState;
+use crate::Model;
+
+crate::routes! {
+    "/search_environment" => {
+        over_write,
+        retrieve_latest,
+    },
+}
+
+editoast_common::schemas! {
+    StdcmSearchEnvironmentCreateForm,
+    StdcmSearchEnvironment,
+}
+
+#[derive(ToSchema)]
+#[cfg_attr(test, derive(Serialize))]
+struct StdcmSearchEnvironmentCreateForm {
+    infra_id: i64,
+    electrical_profile_set_id: Option<i64>,
+    work_schedule_group_id: Option<i64>,
+    timetable_id: i64,
+    search_window_begin: NaiveDateTime,
+    search_window_end: NaiveDateTime,
+}
+
+impl<'de> Deserialize<'de> for StdcmSearchEnvironmentCreateForm {
+    fn deserialize<D>(deserializer: D) -> StdResult<StdcmSearchEnvironmentCreateForm, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        #[serde(deny_unknown_fields)]
+        struct Internal {
+            infra_id: i64,
+            electrical_profile_set_id: Option<i64>,
+            work_schedule_group_id: Option<i64>,
+            timetable_id: i64,
+            search_window_begin: NaiveDateTime,
+            search_window_end: NaiveDateTime,
+        }
+        let internal = Internal::deserialize(deserializer)?;
+
+        // Check dates
+        if internal.search_window_begin >= internal.search_window_end {
+            return Err(SerdeError::custom(format!(
+                "The search environment simulation window begin '{}' must be before the end '{}'",
+                internal.search_window_begin, internal.search_window_end
+            )));
+        }
+
+        Ok(StdcmSearchEnvironmentCreateForm {
+            infra_id: internal.infra_id,
+            electrical_profile_set_id: internal.electrical_profile_set_id,
+            work_schedule_group_id: internal.work_schedule_group_id,
+            timetable_id: internal.timetable_id,
+            search_window_begin: internal.search_window_begin,
+            search_window_end: internal.search_window_end,
+        })
+    }
+}
+
+impl From<StdcmSearchEnvironmentCreateForm> for Changeset<StdcmSearchEnvironment> {
+    fn from(form: StdcmSearchEnvironmentCreateForm) -> Self {
+        StdcmSearchEnvironment::changeset()
+            .infra_id(form.infra_id)
+            .electrical_profile_set_id(form.electrical_profile_set_id)
+            .work_schedule_group_id(form.work_schedule_group_id)
+            .timetable_id(form.timetable_id)
+            .search_window_begin(form.search_window_begin)
+            .search_window_end(form.search_window_end)
+    }
+}
+
+#[utoipa::path(
+    post, path = "",
+    tag = "stdcm_search_environment",
+    request_body = StdcmSearchEnvironmentCreateForm,
+    responses(
+        (status = 201, body = StdcmSearchEnvironment),
+    )
+)]
+async fn over_write(
```

Comment:
Same here, I believe `overwrite` is one word, not 2. That said, if the only way to insert a new `StdcmSearchEnvironment` always overwrite the existing one, I don't see why we have a `retrieve_latest` which would suppose we could e able to get another one?